File: src/test/java/application/browser/MessageHelper.java
package application.browser;

import static org.openqa.selenium.support.ui.ExpectedConditions.invisibilityOfElementLocated;
import static org.openqa.selenium.support.ui.ExpectedConditions.textToBe;

import org.openqa.selenium.By;

public class MessageHelper {
  private MatagBrowser matagBrowser;

  MessageHelper(MatagBrowser matagBrowser) {
    this.matagBrowser = matagBrowser;
  }

  public void hasMessage(String message) {
    matagBrowser.wait(textToBe(By.id("message-text"), message));
  }

  public void hasNoMessage() {
    matagBrowser.wait(invisibilityOfElementLocated(By.id("message")));
  }

  public void close() {
    matagBrowser.findElement(By.id("popup-close")).click();
  }
}

File: src/test/kotlin/application/browser/MessageHelper.kt
package application.browser

import org.openqa.selenium.By
import org.openqa.selenium.support.ui.ExpectedConditions

class MessageHelper internal constructor(private val matagBrowser: MatagBrowser) {
    fun hasMessage(message: String) {
        matagBrowser.wait(ExpectedConditions.textToBe(By.id("message-text"), message))
    }

    fun hasNoMessage() {
        matagBrowser.wait(ExpectedConditions.invisibilityOfElementLocated(By.id("message")))
    }

    fun close() {
        matagBrowser.findElement(By.id("popup-close")).click()
    }
}
